plugin-lifecycle/src/main/java/com/mapbox/maps/plugin/lifecycle/ViewLifecycleOwner.kt

```diff
@@ -0,0 +1,95 @@
+package com.mapbox.maps.plugin.lifecycle
+
+import android.view.View
+import androidx.annotation.UiThread
+import androidx.annotation.VisibleForTesting
+import androidx.annotation.VisibleForTesting.PRIVATE
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleEventObserver
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.LifecycleRegistry
+import androidx.lifecycle.ViewTreeLifecycleOwner
+import java.lang.ref.WeakReference
+import java.util.concurrent.Flow
+
+/**
+ * A [LifecycleOwner] that merges the hosting lifecycle owner's [Lifecycle] (of an Activity or Fragment)
+ * with the [View]'s drawing surface lifecycle.
+ *
+ * This lifecycle is useful for continuously running tasks that obtain the data and draw it in the [View],
+ * for example by collecting [Flow]s and drawing results in the nested views or on the map.
+ *
+ * In general, this lifecycle reflects the hosting lifecycle owner's [Lifecycle] whenever the `View` is attached
```

Comment:
@pengdev this is what I've referred to, main thing fixed for Nav was the coroutines , e.g. with [repeatOnLifecycle](https://developer.android.com/topic/libraries/architecture/coroutines#restart).

I'd probably just highlighted that we keep MapView's lifecycle as CREATED when it is detached, this seem the most important thing for us.

Reply:
will update the inline docs as well.

Reply:
Updated, please take a look.

Reply:
Seems fine to me now